src/main/java/jenkins/plugins/office365connector/BuildListener.java

```diff
@@ -16,7 +16,7 @@
  */
 
 @Extension
-public class Office365ConnectorBuildListener extends RunListener<Run> {
+public class BuildListener extends RunListener<Run> {
```

Comment:
NO! You will only hurt yourself, Extension is just that a class someone can extend.
Please use proper naming of classes, just see how wrong it can go here https://github.com/jenkinsci/configuration-as-code-plugin/pull/811/files#diff-85b0014f902e64b735274e19515eac36R52

fileSystemServiceAccountCredential is the name for a credential type from `kubernetes-credentials` plugin. How on earth would I be able to know that without looking at the source code 😆

Reply:
@casz  thanks for pointing this!